tests/common/get_cts_string.h

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tools for string representation for some types
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_COMMON_GET_CTS_STRING_H
+#define __SYCLCTS_TESTS_COMMON_GET_CTS_STRING_H
+
+#include "common.h"
+
+namespace sycl_cts {
+namespace get_cts_string {
+
+enum class verbosity { brief = 0, detail };
+
+/** @brief Stringify bool value
+ *  @tparam bool flag Value to convert
+ *  @retval String with interpretation of bool
+ */
+inline std::string for_bool(bool flag) { return flag ? "true" : "false"; }
+
+/** @brief Return string's description depending on the type of bundle_state and
+ * verbosity level
+ *  @tparam State bundle state
+ *  @tparam Level verbosity level
+ *  @retval string's description of bundle state
+ */
+template <sycl::bundle_state State, verbosity Level = verbosity::brief>
+inline std::string for_bundle_state() {
+  std::string result;
+  if constexpr (State == sycl::bundle_state::input) {
```

Comment:
I think that if you make the whole function `constexpr` then you can get away with a simple switch statement - you're not guaranteed that optimal code will be generated, but it becomes much more likely. The same for other functions.

Reply:
I see now that the function can't be `constexpr` because of the verbosity level, which is unfortunate. The overhead wouldn't be much IMO, especially compared to the rest of the test. The compiler is also likely to generate a jump table from a switch statement instead of branches anyway. In any case, I don't think this is that important.

Reply:
Thanks for the answer.